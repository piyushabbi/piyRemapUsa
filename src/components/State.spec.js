import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import State from './State';

describe('<State /> Component', () => {
	it('renders State Component', () => {
		const wrapper = shallow(
			<State key="MI" dimensions="M644.5,211" state="MI" fill="#d3d3d3" />
		);
		expect(wrapper.length).toBe(1);
	});

	it('renders with correct props', () => {
		const wrapper = shallow(
			<State key="MI" dimensions="M644.5,211" state="MI" fill="#d3d3d3" />
		);
		// console.log(wrapper.props());
		expect(wrapper.props().d).toBe('M644.5,211');
		expect(wrapper.props().fill).toBe('#d3d3d3');
		expect(wrapper.props()['data-name']).toBe('MI');
	});

	it('should call mouseover event', () => {
		const mouseEventHandler = jest.fn();
		const onMouseOver = jest.fn(() => {
			mouseEventHandler('MI', "Michigan");
		});
		const wrapper = shallow(
			<State 
				mouseEventHandler={onMouseOver}
				state="MI"
				name="Michigan"
			/>
		);
		wrapper.simulate('mouseover');
		expect(onMouseOver).toHaveBeenCalled();
		expect(mouseEventHandler).toHaveBeenCalledWith('MI', "Michigan");
	});

	it('should match snapshot', () => {
		const tree = shallow(
			<State key="MI" dimensions="M644.5,211" state="MI" fill="#d3d3d3" />
		);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
