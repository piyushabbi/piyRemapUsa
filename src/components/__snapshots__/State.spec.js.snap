// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<State /> Component should match snapshot 1`] = `
<path
  className="MI state"
  d="M644.5,211"
  data-name="MI"
  fill="#d3d3d3"
  onMouseOver={[Function]}
/>
`;
